### ISSUE TITLE
Can't use route compilers

### ISSUE THREAD

**Issue by tpow** [MasoniteFramework/masonite#311]
I am trying to use the route compiler for year as described in the documentation. However, either there is something missing in the instructions, I'm missing something, or the `craft serve` command checks routes too early in the process (before the service providers are applied) and fails.

**To Reproduce**
I lifted the example from the documentation here: 
https://docs.masoniteproject.com/v/v2.3/the-basics/routing

Because I had previously added a service provider for a template view composer, I thought I understood what might be needed. However, the example is incomplete.

I ended up creating this in app/providers:
```python
from masonite.provider import ServiceProvider
from masonite.routes import Route


class RouteTypesProvider(ServiceProvider):
    """Provides a Route Compiler for various types"""

    wsgi = False

    def register(self):
        """Register objects into the Service Container"""
        pass

    def boot(self, route: Route):
        route.compile('year', r'([0-9]{4})')
```

I then edited config/providers.py, added the import 
`from app.providers.RouteTypesProvider import RouteTypesProvider`
and added RouteTypesProvider to the end of the PROVIDERS list.

Finally, I added a route that uses the year compiler. In this case I want the parameter also called year, so it looks like this:

`Get('/acct/?year:year', 'AccountController@show')`

I used `craft serve` to test it.

**Expected behavior**
I expected the server to start and be able to use the year compiler.

**Bug**
Despite having used `craft serve` successfully before, it now fails with an error AND worse, acts like the serve command doesn't exist:

```
problem in routes.web - InvalidRouteCompileException: Route compiler "year" is not an available route compiler. Verify you spelled it correctly or that you have added it using the compile() method.

                                   
  [CommandNotFound]                
  Command "serve" is not defined.  
```

**Note**
If I remove the :year from the route, it works again. Do route compilers not work? Did I miss a step? If I did, it would be nice to add this to the documentation. As it was, I had to make a few guesses. (The documentation for View Composers is helpfully more thorough https://docs.masoniteproject.com/v/v2.3/useful-features/view-composers-and-sharing )

**Comment by josephmancuso** [comment]
thanks for bringing this to my attention. I'll take a look tonight 👍

**Comment by josephmancuso** [comment]
@tpow cannot reproduce this. Could you give some more info? Like OS, Masonite version, maybe some screenshots?

**Comment by tpow** [comment]
Thanks for checking. It appears that I simplified the problem too much and didn't check that it was failing everywhere. I'm running this on Ubuntu 18.04.4 LTS bionic, python 3.6, with Masonite==2.3.12.

The year route compiler works in a simple route (as in my original report), but does not work within a Route Group:

```python
    RouteGroup(                                                                  
        [                                                                        
            Get("/accounts/?year:year', 'AccountController@show"),
        ],                                                                       
        prefix="/ledger",                                                          
    ),
```

It looks like the problem occurs when adding the prefix to the RouteGroup. If it is removed (or commented out) then everything works again.

**Comment by tpow** [comment]
I can work around this by not using the prefix at the RouteGroup level and instead adding it to each individual route within the group.

**Comment by josephmancuso** [comment]
Ok thanks I am able to reproduce it when it is inside a route group

**Comment by josephmancuso** [comment]
This should fix that issue https://github.com/MasoniteFramework/masonite/pull/314

One of the issues is that it was compiling in the route group before it was being added via the service provider

**Comment by josephmancuso** [comment]
if you could confirm that fixes it for me before I merge it that would be great

**Comment by tpow** [comment]
I confirm that #314 fixes this for me. `craft show:routes` also works before and after and shows the same routes when the prefix is in the RouteGroup as when adding the prefix to the routes directly. Prior to this fix, show:routes would die when trying to show the routes. Thanks!

**Comment by josephmancuso** [comment]
Fixed in 2.3.13
